DbalConsumerHelperTrait.php

```diff
@@ -134,10 +134,22 @@ protected function removeExpiredMessages(): void
             ->setParameter('redelivered', false, DbalType::BOOLEAN)
         ;
 
-        try {
-            $delete->execute();
-        } catch (RetryableException $e) {
-            // maybe next time we'll get more luck
+        $attempts = 5;
+
+        for ($attempt = 0; $attempt < $attempts; ++$attempt) {
+            $this->getConnection()->beginTransaction();
```

Comment:
It looks like this is all one statement, so there's no need for explicitly starting and committing a transaction.

Reply:
When I read about [root of the issue](https://ph4r05.deadcode.me/blog/2017/12/23/laravel-queues-optimization.html) one of the proposed solutions was:
```
$this->database->transaction(function () use ($queue, $id) {
    if ($this->database->table($this->table)->lockForUpdate()->find($id)) {
        $this->database->table($this->table)->where('id', $id)->delete();
    }
}, 5);
```
I decided to reproduce as much as we can here, this includes transaction. 
I don't mind testing if this will work as expected without transaction, but since I'm not able to reproduce this issue on my own, we'd need another PR that can be checked on QA.